src/ScriptHandler.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace PrestaShop\Composer;
+
+use Composer\Script\Event;
+
+final class ScriptHandler
+{
+    public static function install(Event $event)
+    {
+        $composer = $event->getComposer();
+        $rootPath = realpath($composer->getConfig()->get('vendor-dir') . '/..');
+
+        $extras = $composer->getPackage()->getExtra();
+
+        if (!isset($extras['prestashop'])) {
+            throw new \InvalidArgumentException('The extra.prestashop key needs to be defined.');
+        }
+
+        $config = $extras['prestashop'];
+
+        if (!is_array($config)) {
+            throw new \InvalidArgumentException('The extra.prestashop setting must be an array or a configuration object.');
+        }
+
+        $processor = new ModulesConfigurationProcessor($event->getIO());
```

Comment:
maybe you could give `$rootPath` in the constructor, this would simplify the `processInstallation` and `processUpdate` methods